modules/code-editor/src/backend/api.ts

```diff
@@ -33,6 +33,34 @@ export default async (bp: typeof sdk, editorByBot: EditorByBot) => {
     }
   })
 
+  router.put('/rename/:newName', async (req, res) => {
+    const file = req.body
+    const newName = req.params.newName
+    try {
```

Comment:
When playing with filenames, it can be tricky to use them in an URL. There can be invalid characters, encoding, etc.. It would be a lot better to set the value on the object: { file: oldFile, newName }